app/services/concerns/xccdf/profile_rules.rb

```diff
@@ -7,19 +7,31 @@ module ProfileRules
 
     included do
       def save_profile_rules
+        new_prs, old_prs = profile_rules.partition(&:new_record?)
+
+        # Get rid of all unnecessary connections under a given benchmark
+        ::ProfileRule.joins(:profile)
+                     .where(profile_id: profile_rules.map(&:profile_id))
+                     .where('profiles.parent_profile_id' => nil)
+                     .where.not(id: old_prs)
+                     .delete_all
```

Comment:
First I'd suggest to import new/all and then remove those connections that are extra.

Please do not remove existing. Rather reconcile the extra ones. It has to be done for each profile separately.
Something like
```
profile_rules.group_by(&:profile_id).each do |profile_id, prs|
  ProfileRule.where(profile_id: profile_id)
    .where.not(rule_id: prs.map(&:rule_id))
    .delete_all
end
```
(that could be written with arel in one query, however I'm not sure if it would not hit a param limit)
This call is expected to be run on regular import, hence it does not need to check the canonical state of the profile.

Note that the `.where.not(id: old_prs)` (in the commited code) would not be enough as `old_prs` would hold those rules that are in the benchmark that is being imported. The deletion would lack removal of those extra rule connections.

Reply:
@vkrizan the code above just in addition excludes the `new_prs` from the deletion :thinking: let's talk about this when you're at the office

Reply:
The `old_prs` does not include all existing `ProfileRules` for that benchmark, specifically those that are not on the list of a benchmark that is being imported (those extra ones).

Reply:
What about:
```ruby
     def save_profile_rules
        new_prs, old_prs = profile_rules.partition(&:new_record?)

        new_ids = ::ProfileRule.import!(new_prs, ignore: true)

        # Get rid of all unnecessary connections under a given benchmark
        ::ProfileRule.joins(:profile)
                     .where(profile_id: profile_rules.map(&:profile_id))
                     .where('profiles.parent_profile_id' => nil)
                     .where.not(id: old_prs)
                     .where.not(id: new_ids)
                     .delete_all
      end
```

Reply:
Or is it still necessary to group by `profile_id`? :thinking: